jmclient/jmclient/wallet_service.py

```diff
@@ -441,19 +462,11 @@ def sync_addresses(self):
         jlog.debug("requesting detailed wallet history")
         wallet_name = self.get_wallet_name()
         addresses, saved_indices = self.collect_addresses_init()
-        try:
-            imported_addresses = set(self.bci.rpc('getaddressesbyaccount',
-                                                  [wallet_name]))
-        except JsonRpcError:
-            if wallet_name in self.bci.rpc('listlabels', []):
-                imported_addresses = set(self.bci.rpc('getaddressesbylabel',
-                                                      [wallet_name]).keys())
-            else:
-                imported_addresses = set()
 
-        if not addresses.issubset(imported_addresses):
-            self.bci.add_watchonly_addresses(addresses - imported_addresses,
-                                             wallet_name, self.restart_callback)
+        import_needed = self.bci.import_addresses_if_needed(addresses,
```

Comment:
This rationalisation is very logical (and indeed moves code into bci that should be bci's responsibility, in line with my arch. comments above).